my-app/src/Pages/JoinProject.tsx

```diff
@@ -9,14 +9,38 @@ import Footer from '../Components/Footer'
 import Navbar from '../Components/Navbar';
 import Projects from '../Components/Projects'
 import { useState } from 'react';
-import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
+import { BrowserRouter, Routes, Route, Link, useParams, useLocation} from 'react-router-dom';
+import axios from 'axios';
+import LoginPage from './LoginPage';
+
+interface LocationState {
+    username: string;
+}
+
 
 function Join() {
     const [projectId, setProjectId] = useState('');
-  
+    const location = useLocation()
+    const state = location.state as LocationState
+    const handleChangeProjectId = (event: any) => {
+        setProjectId(event.target.value)
+    }
+    //joinProject/<userId>/<projectId></projectId>
     const handleJoinProject = () => {
-      console.log(`Joining project with ID: ${projectId}`);
-    };
+        console.log("Login clicked")
+        console.log(state.username)
```

Comment:
Getting a null error on reading the username property here

Reply:
Wait there shouldn't be any fontend code in this oopie